google/cloud/spanner/bytes.cc

```diff
@@ -48,10 +48,13 @@ constexpr std::array<unsigned char, UCHAR_MAX + 1> kCharToIndexExcessOne = {{
     38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52,
 }};
 
+// kCharToIndexExcessOne[] assumes an ASCII execution character set.
+static_assert('A' == 65, "required by base64 decoder");
```

Comment:
Note: As far as I can determine, a `static_assert()` must be evaluated in the "execution environment".  For example, if you asserted on the size of a type, it is the execution type you're talking about.  So, the result of this assertion should be affected by something like `-fexec-charset` ... the character set we're compiling for.